streams/test-utils/src/test/java/org/apache/kafka/streams/TopologyTestDriverTest.java

```diff
@@ -897,4 +898,70 @@ public void close() {}
             );
         }
     }
+
+    @Test
+    public void shouldProcessFromSourcesThatMatchMultiplePattern() {
+        final  String patternSourceTopic3 = "source-topic-[A-Z]";
+        final  String consumerTopic3 = "source-topic-Z";
+
+        final ConsumerRecord<byte[], byte[]> consumerRecord3 = consumerRecordFactory.create(consumerTopic3, key1, value1, timestamp1);
+
+        testDriver = new TopologyTestDriver(setupMultipleSourceTopology(PATTERN_SOURCE_TOPIC_1, SOURCE_TOPIC_2, patternSourceTopic3), config);
```

Comment:
You cannot use `setupMultipleSourceTopology` to register a topic-pattern because it calls `addSource(String, String)` but to register a pattern you need to call `addSource(String, Pattern)`

Reply:
@mjsax , Sorry for the previous fix. My bad. I missed due to  copy/paste. I should have double checked it. Committed new changes as per review comments. Kindly review  and let me know if any changes needed.